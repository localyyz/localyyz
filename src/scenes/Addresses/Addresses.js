import React from "react";
import {
  View,
  Text,
  StyleSheet,
  FlatList,
  TouchableHighlight
} from "react-native";

// third party
import { inject, observer } from "mobx-react/native";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";

// custom
import { Colours, Styles, Sizes } from "localyyz/constants";
import { Forms } from "localyyz/components";

@inject(stores => ({
  addresses: stores.addressStore.addresses.slice(),
  fetch: stores.addressStore.fetch
}))
@observer
export default class Addresses extends React.Component {
  static navigationOptions = ({ navigation, navigationOptions }) => ({
    ...navigationOptions,
    title: "Saved Addresses",
    headerRight: (
      <MaterialIcon.Button
        color={"black"}
        backgroundColor={Colours.Transparent}
        onPress={() => navigation.navigate("AddressForm", { title: "Add New" })}
        underlayColor={Colours.Transparent}
        style={{
          justifyContent: "flex-end",
          paddingRight: Sizes.InnerFrame,
          alignItems: undefined
        }}
        iconStyle={{
          marginRight: undefined
        }}
        name="add"
        size={32}
        hitSlop={{
          top: Sizes.OuterFrame,
          bottom: Sizes.OuterFrame,
          left: Sizes.OuterFrame,
          right: Sizes.OuterFrame
        }}/>
    )
  });

  constructor(props) {
    super(props);
    this.settings = this.props.navigation.state.params || {};

    // bindings
    this.renderItem = this.renderItem.bind(this);
    this.onPress = this.onPress.bind(this);
    this.onSelect = this.onSelect.bind(this);
    this.addAddress = this.addAddress.bind(this);
    this.updateAddress = this.updateAddress.bind(this);
  }

  componentDidMount() {
    this.props.fetch();
  }

  renderItem({ item: address }) {
    return (
      <TouchableHighlight
        style={{ width: Sizes.Width }}
        onPress={() => this.onPress(address)}
        underlayColor={Colours.Background}>
        <View style={styles.row}>
          <Text style={styles.rowText}>{address.address}</Text>
          <View style={styles.arrow}>
            <MaterialIcon
              name="keyboard-arrow-right"
              size={Sizes.Text}
              color={Colours.SubduedText}/>
          </View>
        </View>
      </TouchableHighlight>
    );
  }

  onPress(address) {
    !this.onSelect(address) && this.updateAddress(address);
  }

  onSelect(address) {
    if (this.settings.onSelect) {
      this.settings.onSelect(address);

      // and out
      this.props.navigation.goBack(null);

      // help skip updateAddress on onPress
      return true;
    }
  }

  addAddress() {
    return this.props.navigation.navigate("AddressForm", {
      onSubmit: this.onSelect,
      title: "Add New"
    });
  }

  updateAddress(address) {
    return this.props.navigation.navigate("AddressForm", {
      shouldUpdate: true,
      address: address
    });
  }

  addButton = () => {
    return (
      <Forms.Button isEnabled onPress={() => this.addAddress()}>
        Add a new address
      </Forms.Button>
    );
  };

  empty = () => {
    return (
      <View style={styles.footer}>
        <Text style={styles.subtitle}>
          Looks like we don't have an address for you yet.
        </Text>
        {this.addButton()}
      </View>
    );
  };

  render() {
    return (
      <FlatList
        style={styles.container}
        contentContainerStyle={{
          paddingBottom: this.props.addresses.length
            ? Sizes.OuterFrame * 2 + Sizes.IOSTabBar
            : 0
        }}
        data={this.props.addresses}
        keyExtractor={address => `${address.id}`}
        renderItem={this.renderItem}
        ListEmptyComponent={this.empty}
        ItemSeparatorComponent={() => (
          <View
            style={{
              height: 1,
              borderBottomWidth: Sizes.Hairline,
              borderBottomColor: Colours.Border
            }}/>
        )}/>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colours.Foreground
  },

  row: {
    paddingVertical: Sizes.InnerFrame,
    marginHorizontal: Sizes.InnerFrame,
    justifyContent: "space-between",
    alignItems: "center",
    flexDirection: "row",
    height: Sizes.Height / 10
  },

  rowText: {
    ...Styles.Text,
    ...Styles.Emphasized
  },

  subtitle: {
    ...Styles.SectionSubtitle,
    textAlign: "center"
  },

  footer: {
    paddingTop: Sizes.OuterFrame,
    paddingHorizontal: Sizes.OuterFrame
  }
});
