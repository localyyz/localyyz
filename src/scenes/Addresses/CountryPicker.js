import React from "react";
import {
  StyleSheet,
  FlatList,
  Text,
  View,
  TouchableHighlight
} from "react-native";

import { Styles, Sizes, Colours } from "~/src/constants";
import { deviceStore } from "~/src/stores";

import AddressStore from "~/src/stores/AddressStore";

export default class CountryPicker extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      countries: [],
      _options: [],
      _current: [],
      _page: 1,
      _limit: 10
    };

    // must be required props
    this.update = props.navigation.getParam("update");
  }

  componentDidMount = () => {
    const deviceCountryCode = deviceStore.getDeviceCountry();
    AddressStore.fetchCountries().then(resolved => {
      if (resolved.countries) {
        let countries = [];
        for (let country of resolved.countries) {
          let c = {
            value: country.alpha2Code,
            name: country.name,
            nativeName: country.nativeName,
            key: country.alpha2Code
          };

          if (c.key == deviceCountryCode) {
            countries.unshift({
              ...c,
              isSuggested: true,
              description: "Suggested country:"
            });
          } else {
            countries.push(c);
          }
        }

        this.setState({
          countries: countries,
          _options: countries.slice(0, this.state._limit)
        });
      }
    });
  };

  onEndReached = () => {
    let start = this.state._page * this.state._limit;
    let end = start + this.state._limit;

    if (start > this.state.countries.length) {
      return;
    }

    this.setState({
      _options: this.state._options.concat(
        this.state.countries.slice(start, end)
      ),
      _page: this.state._page + 1
    });
  };

  onSelect = value => {
    this.update(value);
  };

  renderItem = ({ item: country }) => {
    const label
      = country.name == country.nativeName ? country.name : country.nativeName;

    return (
      <TouchableHighlight
        style={{ width: Sizes.Width }}
        onPress={() =>
          this.onSelect({
            country: country.nativeName,
            countryCode: country.value
          })
        }
        underlayColor={Colours.Background}>
        <View
          style={[styles.row, country.isSuggested ? styles.suggestedRow : {}]}>
          {country.description ? (
            <Text style={styles.rowDescription}>{country.description}</Text>
          ) : null}
          <Text style={styles.rowText}>{label}</Text>
        </View>
      </TouchableHighlight>
    );
  };

  render() {
    return (
      <FlatList
        renderItem={this.renderItem}
        keyExtractor={o => o.key}
        scrollEventThrottle={16}
        maxToRenderPerBatch={10}
        initialNum={10}
        removeClippedSubviews={true}
        data={this.state._options}
        contentContainerStyle={styles.container}
        onEndReached={this.onEndReached}
        ItemSeparatorComponent={() => (
          <View
            style={{
              height: 1,
              borderBottomWidth: Sizes.Hairline,
              borderBottomColor: Colours.Border
            }}/>
        )}/>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colours.Foreground,
    paddingTop: Sizes.OuterFrame,
    paddingBottom: Sizes.IOSTabBar + Sizes.OuterFrame * 2
  },

  row: {
    paddingVertical: Sizes.InnerFrame,
    marginHorizontal: Sizes.InnerFrame,
    justifyContent: "center",
    height: Sizes.Height / 10
  },

  suggestedRow: {
    marginBottom: Sizes.OuterFrame
  },

  rowText: {
    ...Styles.Text,
    ...Styles.Emphasized
  },

  rowDescription: {
    ...Styles.SmallText,
    color: Colours.SubduedText,
    paddingBottom: 5
  }
});
