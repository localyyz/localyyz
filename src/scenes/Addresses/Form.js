import React from "react";
import {
  Keyboard,
  KeyboardAvoidingView,
  SafeAreaView,
  View,
  StyleSheet,
  ScrollView
} from "react-native";

// third party
import { Provider, observer, inject } from "mobx-react/native";

// custom
import { Styles, Sizes, Colours } from "localyyz/constants";
import { Forms } from "localyyz/components";
import { UserAddress } from "localyyz/models";
import { navbarStore } from "~/src/stores";

import GoogleAddress from "./GoogleAddress";

// constants
const DEBUG = false;
const SERVER_FAILURE
  = "There was an internal server error and your request could not be processed";

@inject(stores => ({
  add: address => stores.addressStore.add(address),
  update: address => stores.addressStore.update(address),
  delete: address => stores.addressStore.remove(address.id)
}))
@observer
export default class AddressForm extends React.Component {
  static navigationOptions = ({ navigation, navigationOptions }) => ({
    ...navigationOptions,
    title: navigation.getParam("title", "Edit")
  });

  constructor(props) {
    super(props);
    this.settings
      = (this.props.navigation
        && this.props.navigation.state
        && this.props.navigation.state.params)
      || {};
    this.store = new Forms.Store(
      // internally used
      "id",

      // associated form fields
      {
        id: "firstName",
        validators: ["isRequired"]
      },
      { id: "lastName", validators: ["isRequired"] },
      { id: "street", validators: ["isRequired"] },
      "addressOpt",
      { id: "city", validators: ["isRequired"] },
      { id: "region" },
      { id: "postal", validators: ["isRequired"] },
      {
        id: "country",
        validators: ["isRequired"],
        options: {}
      },
      {
        id: "address",

        // mapping UserAddress fields to these fields on update
        mapping: {
          address: "street",
          city: "city",
          province: "region",
          zip: "postal"
        }
      }
    );
    this.state = { manual: false };
  }

  componentDidMount() {
    // merge in outside data if provided, if address was never touched,
    // allowing address component field syncing break it up into
    // all the associated fields
    if (this.settings.address) {
      for (let key in this.settings.address) {
        // NOTE: there are some weird mapping thing going on here, for
        // now just blindly support it
        let mappingKey = "";
        switch (key) {
          case "address":
            mappingKey = "street";
            break;
          case "province":
            mappingKey = "region";
            break;
          case "zip":
            mappingKey = "postal";
            break;
          default:
            mappingKey = key;
        }
        this.store.update(mappingKey, this.settings.address[key]);
      }
    }
  }

  submit = async () => {
    try {
      // attempt update or add
      let address = await (this.shouldUpdate
        ? this.props.update
        : this.props.add)(this.address);

      // bail prematurely due to server failure
      // TODO: call setError for field errors on status 422
      if (address.hasError) {
        throw SERVER_FAILURE;
      }

      // if callback given
      this.settings.onSubmit
        && this.settings.onSubmit(new UserAddress(address));

      // alternative to navigation
      if (this.settings.onComplete) {
        this.settings.onComplete();
      } else {
        this.props.navigation.goBack(null);
      }
    } catch (err) {
      DEBUG && console.log("AddressForm: Failed to set address.", err);
      navbarStore.notify(`${err} - Please try again later`, {
        backgroundColor: Colours.Fail
      });
    }
  };

  delete = async () => {
    try {
      let response = await this.props.delete(this.address);
      if (response.hasError) {
        throw SERVER_FAILURE;
      }

      // if callback given
      this.settings.onDelete && this.settings.onDelete();

      // alternative to navigation
      if (this.settings.onComplete) {
        this.settings.onComplete();
      } else {
        this.props.navigation.goBack();
      }
    } catch (err) {
      DEBUG && console.log("AddressForm: Failed to delete address.", err);
      navbarStore.notify(`${err} - Please try again later`, {
        backgroundColor: Colours.Fail
      });
    }
  };

  get shouldUpdate() {
    return !!this.address && !!this.address.id;
  }

  get submitButton() {
    return (
      this.settings.buttonLabel
      || (this.shouldUpdate ? "Update this address" : "Add this address")
    );
  }

  get deleteButton() {
    return this.settings.deleteButtonLabel || "Remove this address";
  }

  get title() {
    return this.settings.title || this.shouldUpdate
      ? "Update address details"
      : "Add a new address";
  }

  get address() {
    // these are basically reverse of the above merge object, should be in
    // sync with the mapping above
    return this.store.isComplete
      ? new UserAddress({
          id: this.store.data.id,
          address: this.store.data.street,
          city: this.store.data.city,
          province: this.store.data.region,
          zip: this.store.data.postal,
          country: this.store.data.country,
          firstName: this.store.data.firstName,
          lastName: this.store.data.lastName,
          addressOpt: this.store.data.addressOpt
        })
      : null;
  }

  pickCountry = () => {
    this.props.navigation.navigate("CountryPicker", {
      field: "country",
      title: "Select a Country",
      update: value => this.store.update("country", value)
    });
  };

  onSubmitEditing = () => {
    const nextField = this.store.nextField;
    if (nextField) {
      switch (nextField.id) {
        case "firstName":
          this._firstNameRef.focus();
          break;
        case "lastName":
          this._lastNameRef.focus();
          break;
        case "city":
          this._cityRef.focus();
          break;
        case "street":
          this._streetRef.focus();
          break;
        case "postal":
          this._postalRef.focus();
          break;
        case "country":
          this._countryRef.focus();
          break;
      }
    } else {
      Keyboard.dismiss();
    }
  };

  renderManualForm = () => {
    return (
      <KeyboardAvoidingView style={styles.container} behavior="padding" enabled>
        <ScrollView contentContainerStyle={styles.content}>
          <View style={Styles.Horizontal}>
            <Forms.Field
              inputRef={ref => (this._firstNameRef = ref)}
              autoFocus={true}
              onSubmitEditing={this.onSubmitEditing}
              field="firstName"
              label="First name"
              autoCorrect={false}
              style={styles.largerField}/>
          </View>
          <View style={Styles.Horizontal}>
            <Forms.Field
              inputRef={ref => (this._lastNameRef = ref)}
              onSubmitEditing={this.onSubmitEditing}
              field="lastName"
              label="Last name"
              autoCorrect={false}
              style={styles.largerField}/>
          </View>
          <View>
            <View style={Styles.Horizontal}>
              <Forms.Field
                inputRef={ref => (this._streetRef = ref)}
                onSubmitEditing={this.onSubmitEditing}
                field="street"
                label="Address"
                style={styles.largerField}/>
            </View>
            <View style={Styles.Horizontal}>
              <Forms.Field
                field="addressOpt"
                label="Optional Apt, Suite or Floor"
                style={styles.largerField}
                onSubmitEditing={this.onSubmitEditing}/>
            </View>
            <Forms.Field
              inputRef={ref => (this._cityRef = ref)}
              field="city"
              label="City"
              onSubmitEditing={this.onSubmitEditing}/>
            <View style={Styles.Horizontal}>
              <Forms.Field
                field="region"
                label="State/Province/Region"
                onSubmitEditing={this.onSubmitEditing}
                style={styles.largerField}/>
              <Forms.Field
                inputRef={ref => (this._postalRef = ref)}
                field="postal"
                label="Postal Code"
                onSubmitEditing={this.onSubmitEditing}/>
            </View>
            <Forms.Field
              inputRef={ref => (this._countryRef = ref)}
              onPress={this.pickCountry}
              onFocus={this.pickCountry}
              field="country"
              label="Country"/>
          </View>
          <View style={styles.buttons}>
            <Forms.Button onPress={() => this.submit()}>
              {this.submitButton}
            </Forms.Button>
            {this.shouldUpdate ? (
              <Forms.Button
                isEnabled
                onPress={() => this.delete()}
                color={Colours.DisabledButton}
                labelColor={Colours.Text}>
                {this.deleteButton}
              </Forms.Button>
            ) : null}
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    );
  };

  onPressManual = () => {
    this.setState({ manual: true });
  };

  render() {
    return (
      <Provider formStore={this.store}>
        <SafeAreaView style={styles.container}>
          {this.store.data.street || this.state.manual ? (
            this.renderManualForm()
          ) : (
            <GoogleAddress
              field={"address"}
              onPressManual={this.onPressManual}/>
          )}
        </SafeAreaView>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colours.Foreground
  },

  content: {
    backgroundColor: Colours.Foreground,
    paddingTop: Sizes.OuterFrame,
    paddingBottom: Sizes.IOSTabBar + Sizes.OuterFrame * 2
  },

  largerField: {
    flex: 1
  },

  buttons: {
    marginTop: Sizes.InnerFrame
  }
});
