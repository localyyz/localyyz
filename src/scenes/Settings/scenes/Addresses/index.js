import React from "react";
import { View, StyleSheet, FlatList } from "react-native";

// third party
import { inject, observer } from "mobx-react/native";

// custom
import { Sizes } from "localyyz/constants";
import { BaseScene, Forms } from "localyyz/components";

@inject(stores => ({
  addresses: stores.addressStore.addresses,
  fetch: stores.addressStore.fetch
}))
@observer
export default class Addresses extends React.Component {
  constructor(props) {
    super(props);
    this.settings = this.props.navigation.state.params || {};

    // bindings
    this.renderItem = this.renderItem.bind(this);
    this.onPress = this.onPress.bind(this);
    this.onSelect = this.onSelect.bind(this);
    this.addAddress = this.addAddress.bind(this);
    this.updateAddress = this.updateAddress.bind(this);
    this.fetch = this.fetch.bind(this);
  }

  componentDidMount() {
    this.fetch();
  }

  async fetch() {
    let addresses = await this.props.fetch();

    // forward to add address by default if none exist
    if (!addresses || addresses.length < 1) {
      this.addAddress();
    }
  }

  renderItem({ item: address }) {
    return (
      <Forms.BaseField
        label={address.address}
        onPress={() => this.onPress(address)}/>
    );
  }

  onPress(address) {
    !this.onSelect(address) && this.updateAddress(address);
  }

  onSelect(address) {
    if (this.settings.onSelect) {
      this.settings.onSelect(address);

      // and out
      this.props.navigation.goBack(null);

      // help skip updateAddress on onPress
      return true;
    }
  }

  addAddress() {
    return this.props.navigation.navigate("AddressForm", {
      onSubmit: this.onSelect
    });
  }

  updateAddress(address) {
    return this.props.navigation.navigate("AddressForm", {
      shouldUpdate: true,
      address: address
    });
  }

  get footer() {
    return (
      <View style={styles.footer}>
        <Forms.Button isEnabled onPress={() => this.addAddress()}>
          Add a new address
        </Forms.Button>
      </View>
    );
  }

  render() {
    return (
      <BaseScene backAction={this.props.navigation.goBack} title="Addresses">
        <View style={styles.content}>
          <FlatList
            scrollEnabled={false}
            data={this.props.addresses && this.props.addresses.slice()}
            keyExtractor={address => `${address.id}`}
            renderItem={this.renderItem}
            ListFooterComponent={this.footer}/>
        </View>
      </BaseScene>
    );
  }
}

const styles = StyleSheet.create({
  content: {
    flex: 1
  },

  footer: {
    alignItems: "center",
    marginVertical: Sizes.InnerFrame,
    marginHorizontal: Sizes.OuterFrame
  }
});
