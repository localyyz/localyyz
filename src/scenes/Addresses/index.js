import React from "react";
import { HeaderBackButton, createStackNavigator } from "react-navigation";

// local
import Addresses from "./Addresses";
import Form from "./Form";
import CountryPicker from "./CountryPicker";

export default createStackNavigator(
  {
    Addresses: Addresses,
    AddressForm: Form,
    CountryPicker: CountryPicker
  },
  {
    navigationOptions: ({ navigation, navigationOptions }) => ({
      ...navigationOptions,
      headerLeft: (
        <HeaderBackButton
          tintColor="#000"
          onPress={() => navigation.goBack(null)}/>
      )
    })
  }
);
