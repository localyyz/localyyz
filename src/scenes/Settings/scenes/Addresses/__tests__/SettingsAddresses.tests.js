import React from "react";
import Renderer from "react-test-renderer";
import Addresses from "../index";

const PROPS = {
  navigation: {
    navigate: jest.fn(),
    state: {
      params: "test"
    }
  },
  fetch: jest.fn(),
  addresses: [
    { id: 1, address: "180 John Street West" },
    { id: 2, address: "120 John Street East" }
  ]
};
describe("Settings Addresses", () => {
  it("Settings Addresses: should render properly", () => {
    const rendered = Renderer.create(<Addresses.wrappedComponent {...PROPS} />);
    expect(rendered.getInstance()).toBeDefined();
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
