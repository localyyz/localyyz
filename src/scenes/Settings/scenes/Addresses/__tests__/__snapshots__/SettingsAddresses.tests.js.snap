// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Addresses Settings Addresses: should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 2,
          },
          Object {
            "opacity": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 15,
            "marginTop": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 50,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "borderBottomColor": "#FFFFFF",
                  "color": "#FFFFFF",
                  "fontFamily": "Menlo-Regular",
                  "fontSize": 16,
                  "fontWeight": "300",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            ADDRESSES
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
      >
        <View
          constrainWidth={750}
          source={
            Object {
              "uri": undefined,
            }
          }
        >
          <View>
            <BVLinearGradient
              colors={
                Array [
                  16777215,
                  16777215,
                  16777215,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              style={
                Object {
                  "paddingBottom": 15,
                  "paddingHorizontal": 15,
                  "paddingTop": 75,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 22,
                      "fontWeight": "700",
                    },
                    undefined,
                  ]
                }
              >
                Addresses
              </Text>
            </BVLinearGradient>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          onScroll={[Function]}
          scrollEventThrottle={16}
          showsVerticalScrollIndicator={false}
        >
          <View>
            <View
              style={
                Object {
                  "paddingTop": 1334,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingBottom": 100,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <RCTScrollView
                  data={
                    Array [
                      Object {
                        "address": "180 John Street West",
                        "id": 1,
                      },
                      Object {
                        "address": "120 John Street East",
                        "id": 2,
                      },
                    ]
                  }
                  disableVirtualization={false}
                  getItem={[Function]}
                  getItemCount={[Function]}
                  horizontal={false}
                  initialNumToRender={10}
                  keyExtractor={[Function]}
                  maxToRenderPerBatch={10}
                  numColumns={1}
                  onContentSizeChange={[Function]}
                  onEndReachedThreshold={2}
                  onLayout={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  renderItem={[Function]}
                  scrollEnabled={false}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  updateCellsBatchingPeriod={50}
                  viewabilityConfigCallbackPairs={Array []}
                  windowSize={21}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "flexBasis": 150,
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#FFFFFF",
                                "paddingHorizontal": 15,
                                "paddingVertical": 10,
                              },
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginRight": 15,
                              }
                            }
                          >
                            <View
                              style={Object {}}
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "rgba(255, 255, 255, 0)",
                                      "color": "rgb(30, 30, 30)",
                                      "fontSize": 14,
                                      "fontWeight": "300",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                180 John Street West
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "flex-end",
                                  "flex": 1,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#999999",
                                      "fontSize": 16,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "Material Icons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "flexBasis": 150,
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#FFFFFF",
                                "paddingHorizontal": 15,
                                "paddingVertical": 10,
                              },
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginRight": 15,
                              }
                            }
                          >
                            <View
                              style={Object {}}
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "rgba(255, 255, 255, 0)",
                                      "color": "rgb(30, 30, 30)",
                                      "fontSize": 14,
                                      "fontWeight": "300",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                120 John Street East
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "flex-end",
                                  "flex": 1,
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={false}
                                ellipsizeMode="tail"
                                style={
                                  Array [
                                    Object {
                                      "color": "#999999",
                                      "fontSize": 16,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "Material Icons",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                  ]
                                }
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "marginHorizontal": 25,
                      "marginVertical": 15,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "marginTop": 15,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#9013FE",
                              "borderRadius": 15,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "paddingHorizontal": 30,
                              "paddingVertical": 7.5,
                            },
                            false,
                            undefined,
                          ]
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "backgroundColor": "rgba(255, 255, 255, 0)",
                                "borderBottomColor": "#FFFFFF",
                                "color": "#FFFFFF",
                                "fontSize": 16,
                                "fontWeight": "600",
                              },
                              false,
                              undefined,
                            ]
                          }
                        >
                          Add a new address
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
